### DISCUSSION TITLE
Ctrl-S for incremental search: does it work at all?

### DISCUSSION THREAD

**Discussion by dbikel** [whitphx/vscode-emacs-mcx#1332] [Q&A]
Is there something special that you need to do to get incremental search to work?  I found that ctrl-S was initially bound both to the native "Find" command (that opens up the native search widget) as well as to `emacs-mcx.isearchForward` command.  While you can, indeed, do a search using the native VSCode Find command, it is not incremental search.

**Comment by kshartman** [comment]
when you look at the keyboard shortcuts check the when clauses.  you can have more than one command bound to a command.  they will execute in different contexts depending on the when clause.  This extension's iSearch works perfectly for me (at least in javascript language).  When I look at my bindings on Ctrl+S is only bound to Emasc-MCX: Isearch Forward,  If you have it bound to that and something else with different when clauses then its coming from different extensions is my bet.  You can tell where it comes form by examining the source column in Keyboard Shortcuts.  If you do have multiple bindings, I suggest you remove the offending extension or edit its keybinding. I have attached my bindings for Ctrl-S for reference.
![ctrls](https://user-images.githubusercontent.com/4327353/235981470-c1dac0b4-c343-40ee-a3a9-d89458cc4c38.png)
